src/textual/widgets/_tabbed_content.py

```diff
@@ -422,38 +534,77 @@ def _is_associated_tabs(self, tabs: Tabs) -> bool:
         """
         return isinstance(tabs, ContentTabs) and tabs.tabbed_content is self
 
-    def _watch_active(self, active: str) -> None:
+    def _watch_active(self, old_active: str, active: str) -> None:
         """Switch tabs when the active attributes changes."""
         with self.prevent(Tabs.TabActivated):
-            self.get_child_by_type(ContentTabs).active = active
+            self.get_child_by_type(ContentTabs).active = ContentTab.add_prefix(active)
             self.get_child_by_type(ContentSwitcher).current = active
 
     @property
     def tab_count(self) -> int:
         """Total number of tabs."""
         return self.get_child_by_type(ContentTabs).tab_count
 
+    def get_tab(self, pane_id: str | TabPane) -> Tab:
+        """Get the `Tab` associated with the given ID or `TabPane`.
+
+        Args:
+            pane_id: The ID of the pane, or the pane itself.
+
+        Returns:
+            The Tab associated with the ID.
+        """
+        if target_id := pane_id if isinstance(pane_id, str) else pane_id.id:
```

Comment:
I'd suggest parenthesising the conditional expression to make it easier to read inside the `if` statement.

```suggestion
        if target_id := (pane_id if isinstance(pane_id, str) else pane_id.id):
```